src/main/scala/advancepassengerinfo/importer/DqApiFeed.scala

```diff
@@ -35,9 +37,11 @@ case class DqApiFeedImpl(fileNamesProvider: FileNames,
         fileProcessor.process(zipFileName)
           .map {
             case Some((total, successful)) =>
+              processState.update()
```

Comment:
If we read the last processed time from the database then we don't need to manually call a function to update it here

Reply:
If we take date from database and it is not updated for few hours due to file not being present then do we say health check pass ? as in this we are updating processState in case `None` as well .  which mean process is still checking but no files at moment .

Reply:
The place you're calling `update()` from is when a file has been either successfully or unsuccessfully processed:
```
fileProcessor.process(zipFileName)
          .map {
            case Some((total, successful)) =>
              processState.update()
              log.info(s"$successful / $total manifests successfully processed from $zipFileName")
              metricsCollector.counter("api-dq-manifests-processed", successful)
            case None =>
              processState.update()
              log.info(s"$zipFileName could not be processed")
              metricsCollector.counter("api-dq-zip-failure", 1)
          }
          .map(_ => zipFileName)
```
Maybe that's not where you intended to put it?